File: server-status.js
document.addEventListener("DOMContentLoaded", async function() {
    const statusIndicator = document.querySelector(".status-indicator");
    const onlineCount = document.querySelector(".online-count");

    try {
        const response = await fetch("server-status.py");
        const data = await response.json();

        if (data.onlinePlayers > 0) {
            statusIndicator.textContent = "Online";
            statusIndicator.classList.add("online");
        } else {
            statusIndicator.textContent = "Offline";
            statusIndicator.classList.add("offline");
        }

        onlineCount.textContent = data.onlinePlayers;
    } catch (error) {
        statusIndicator.textContent = "Error";
        statusIndicator.classList.add("error");
        onlineCount.textContent = "N/A";
    }
});

File: server-status.py
from mcstatus import MinecraftServer
import json

# Replace with your Minecraft server IP
server_ip = "eonverse.club"

server = MinecraftServer.lookup(server_ip)

try:
    status = server.status()
    online_players = status.players.online
    motd = status.description
except:
    online_players = 0
    motd = "Server Offline"

data = {
    "onlinePlayers": online_players,
    "motd": motd
}

print(json.dumps(data))
